Patch 5

mukhlesur-millon opened the pull request:
**add new arguments -S, --save-links-path
*add new assetType presentation
*add ``# -*- coding: utf-8 -*-`` according to 'pep-0263' this fix unicode problem , tested on 'https://www.udemy.com/kylechinese2/' in 'kali sana'

nishad commented:
* PEP0263, deals with the encoding of the source code, It is used for the editors or the python interpreter. It doesn't solve Unicode related issues in programming level.

* We already got a `-o` argument. I'm more favorable to reuse that instead creating a new one as a PEP20 approach. 

* https://github.com/mukhlesur-millon/udemy-dl/blob/bd688422cc5e65799686f80b423641db4971e04b/src/udemy_dl/udemy_dl.py#L149, seems a bit cluttered, we can think of a list here, and more programmatic sustainable approach to proceed. However, you can submit a pull request only for the presentation part.

mukhlesur-millon commented:
thanks, now i have got ``PEP0263``, it works with editor when i save a file. my test worked because i had merge it with your master. i have deleted these line.

``-S`` has nothing to do with ``-o`` ;  choosing path and name of saved links #21 
i will use here ``udemy-dl -s -o C:/udemy-dl/links.txt http://udemy.com/***``
instead of ``udemy-dl -S C:/udemy-dl/links.txt http://udemy.com/***``  if you think it is better.

i have changed line 149 to a list.

nishad commented:
I think the implementation of presentation is not complete. What kind of extension they are using ? And how it is going to be dealt in the downloaders here https://github.com/mukhlesur-millon/udemy-dl/blob/db105d61426e163108d614b32a47061973679d0b/src/udemy_dl/download.py#L13 ?

If we got an output switch, we are going to use that for every output. Irrespective of the format of the output. So accepting more and more output switches for every feature is not acceptable. The same is applicable for '-s'.

Quality of the code is much important than any temporary gain we are getting from it. So we need to stick to the standards. Here we follow [PEP0020](https://www.python.org/dev/peps/pep-0020/).

mukhlesur-millon commented:
i meet with python for may be one month even i haven't use github though i had some forks. so it is harder for me to follow every roles of python. if i do any mistake then i am expecting your suggestion and modification. 

i will change -S soon.

when -s output a links.txt file, it has nothing to do with ``def download(link, filename):``. one have to use a external downloader like idm. it has batch downloader mode and queue mode. i had written a script to download with idm. i will add it here.  they will use extension from filename.

nishad commented:
Its not about removing `-s`, its all about reusing `-o` for the output location.

mukhlesur-millon commented:
i am removing -S not -s.

nishad commented:
I was asking about the downloading of presentation type files. Sorry if I made you confused.

> *I think the implementation of presentation is not complete. What kind of extension they are using ? And how it is going to be dealt in the downloaders here https://github.com/mukhlesur-millon/udemy-dl/blob/db105d61426e163108d614b32a47061973679d0b/src/udemy_dl/download.py#L13 ?*

mukhlesur-millon commented:
please made me clear what you meant by  implementation of presentation , using of entension and dealing with downloader.
